### PR TITLE
PHPC-2267: Support BSON Binary subtype 0x08 (sensitive)

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/PHPC-2267

I saw this was included in libmongoc 1.25, so I think it makes sense to throw into 1.17.

Docs: https://github.com/php/doc-en/pull/2939

### CODE REVIEW

File: tests/bson-corpus/binary-valid-014.phpt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+$canonicalExtJson = '{"x" : { "$type" : {"$numberInt": "2"}}}';
+
+// Canonical BSON -> Native -> Canonical BSON
+echo bin2hex(fromPHP(toPHP($canonicalBson))), "\n";
+
+// Canonical BSON -> BSON object -> Canonical BSON
+echo bin2hex((string) MongoDB\BSON\Document::fromBSON($canonicalBson)), "\n";
+
+// Canonical BSON -> Canonical extJSON
+echo json_canonicalize(toCanonicalExtendedJSON($canonicalBson)), "\n";
```
Comment (GromNaN) [inline tests/bson-corpus/binary-valid-014.phpt:22]: I think `json_canonicalize` is not needed for the tested data.

I've seen the `json_canonicalize` function have a patch for PHP < 7.1 that could be removed.

File: tests/bson-corpus/binary-valid-011.phpt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 --TEST--
-Binary type: subtype 0x80
```
Comment (GromNaN) [inline tests/bson-corpus/binary-valid-011.phpt:2]: Why did you used this code that isn't in the list?

Comment (GromNaN) [reply tests/bson-corpus/binary-valid-011.phpt:2]: Just seen it in the tests corpus. You simply renamed some test files.

**Review by GromNaN — APPROVED** [review]

File: tests/bson-corpus/binary-valid-014.phpt (hunk 1 above)
Comment (jmikola) [reply tests/bson-corpus/binary-valid-014.phpt:22]: Created [PHPC-2320](https://jira.mongodb.org/browse/PHPC-2320) to remove the string replacement logic. The function is still relevant for `$numberDouble`.

File: tests/bson-corpus/binary-valid-011.phpt (hunk 2 above)
Comment (jmikola) [reply tests/bson-corpus/binary-valid-011.phpt:2]: This is due to https://github.com/mongodb/specifications/commit/db7eb39bba5eafa33f48376dbe224180c8af705e adding a test in the middle of an array. That causes [convert-bson-corpus-tests.php](https://github.com/mongodb/mongo-php-driver/blob/master/scripts/convert-bson-corpus-tests.php) to shift all of the generated `phpt` files from that point forward.

**Review by alcaeus — APPROVED** [review]

**Pull request merged**